[Jaehyeok Choi] iP

choiwab opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

choiwab committed:
Rename chatbot in Duke.java

choiwab committed:
Add Echo to Duke.java

choiwab committed:
add list to Duke.java

choiwab committed:
Add Task.java

choiwab committed:
Add marking to Duke.java

choiwab committed:
Add Deadline.java Event.java Todo.java

choiwab committed:
ToDos, Events, Deadlines in Duke.java

choiwab committed:
Add BlacknutExceptions.java

choiwab committed:
Configure package for exception class and exception handling

choiwab committed:
Add Delete to Blacknut.java

choiwab committed:
Implement Level 7: Save and load tasks from disk

choiwab committed:
Merge branch-Level-7 into master

choiwab committed:
Implement Level 8: Time

choiwab committed:
Merge branch-Level-8 into master

choiwab committed:
Implement Level 9: Find

choiwab committed:
Merge branch-Level-9 into master

choiwab committed:
Implement Packages

choiwab committed:
Package merge

choiwab committed:
Implement Level 8: Reformation of date formats

choiwab committed:
Merge branch-Level-8 into master

choiwab committed:
Implement OOP

choiwab committed:
Merge A-OOP to master

jhqiu21 commented on src/main/java/blacknut/ui/Blacknut.java:
Please avoid commenting out large blocks of code.

jhqiu21 reviewed (commented):
Overall, LGTM. Your code easy to read and use  quite naming style. Just a few nits to fix.

choiwab committed:
Add gradle to project

choiwab committed:
Merge Gradle into master

t-leongchuan commented on src/main/java/blacknut/ui/Storage.java:
```diff
@@ -0,0 +1,53 @@
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> loadTasksFromFile() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        File file = new File(filePath);
+        if (!file.exists()) {
+            return tasks;
+        }
+
+        try {
+            java.io.BufferedReader reader = new java.io.BufferedReader(new java.io.FileReader(filePath));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                Task task = Task.fromFileFormat(line);
+                tasks.add(task);
+            }
+            reader.close();
+        } catch (java.io.IOException | BlacknutExceptions.IncorrectFormatException e) {
+            System.out.println("Error loading tasks from file: " + e.getMessage());
+        }
+        return tasks;
+    }
```
LGTM. Overall, code follows coding standard and code is well structured. Well Done! 👍

t-leongchuan replied:
LGTM. Overall, code follows coding standard and code is well structured. Well Done! 👍

choiwab committed:
Implement JUnit test cases

choiwab committed:
Merge A-JUnit branch to master

choiwab committed:
Implement JavaDoc

choiwab committed:
Merge branch-A-JavaDoc into Master

choiwab committed:
Implement checkstyle

choiwab committed:
Merge checkstyle into master

choiwab committed:
Implement JavaFX

choiwab committed:
Merge GUI branch to master

brianchoon commented on data/blacknut.txt:
Wondering if you intended to track the version history of this file? If you intended to use it solely for testing, you can .gitignore it!

brianchoon commented on src/main/java/blacknut/ui/Blacknut.java:
```diff
@@ -0,0 +1,340 @@
+//            throw new InvalidTaskNumberException("Task number out of range. Please provide a valid number from the list.");
+//        }
+//    }
+//}
```
Could I clarify what this really long chunk of code (200+) commented out is doing? Consider moving it to another file if you really need to keep a backup of past code.

brianchoon commented on src/main/java/blacknut/ui/Blacknut.java:
```diff
@@ -0,0 +1,340 @@
+            try {
+                String command = parser.parseCommand(input);
+                switch (command) {
+                    case "bye":
```
Remember that as per the coding standard, there is no indentation for case clauses :)

brianchoon commented on src/main/java/blacknut/ui/Parser.java:
```diff
@@ -0,0 +1,93 @@
+    public String[] parseEvent(String input) throws BlacknutExceptions.IncorrectFormatException {
+        String[] parts = input.substring(6).split(" /from | /to ");
+        if (parts.length != 3 || parts[0].trim().isEmpty() || parts[1].trim().isEmpty() || parts[2].trim().isEmpty()) {
+            throw new BlacknutExceptions.IncorrectFormatException("The format for an event should be: event <description> /from <yyyy-mm-dd> /to <yyyy-mm-dd>");
```
nit: Lines should be kept to at most 120 chars!

brianchoon commented on src/main/java/blacknut/ui/Task.java:
```diff
@@ -0,0 +1,105 @@
+
+        Task task;
+        switch (type) {
+            case "T":
```
Same here: case should not be indented

brianchoon reviewed (commented):
Hello, your code is written really well, and the appropriate classes are used too! Do take note of the coding standards to follow, overall well done!

choiwab committed:
Add assertion checks for task validity and input data

This commit adds assertion checks across the Task class to catch invalid states and inputs early during runtime, improving code reliability. \n\n Let's *Constructor: Ensures task description is neither null nor empty when creating a new task.\n fromFileFormat method:Asserts that the input string is not null or empty before parsing.\n *toFileFormat method: Adds checks to ensure the task description is valid before converting the task to a savable format.\n*markAsDone and markAsNotDone methods: Introduces assertions to ensure these methods are used only when applicable (e.g., not marking a task as done if already done).\n*toString method: Ensures the task description is valid when generating a string representation.\n\n By introducing these assertions, we prevent null pointers, incorrect task formats, and improve the overall stability of the application

choiwab committed:
Merge pull request #1 from choiwab/branch-A-Assertions

Add assertion checks for task validity and input data

choiwab committed:
Remove obsolete code form Blacknut.java for improved code quality

The code removed from Blacknut.java was outdated and no longer relevant to the current functionality of the project. Keeping this obsolete code could lead to confusion, maintenance challenges, and potential errors during future development.

Removing the commented out code improves the readability and quality of the code

Let's delete out all the commented out code from the Blacknut.java file

:wq

choiwab committed:
Integrate Blacknut Messages to UI

choiwab committed:
Merge pull request #2 from choiwab/A-CodeQuality

A code quality

choiwab committed:
Adjust for Better Code Quality

Case clauses must have no indentations and a line must not have more than 120 characters

A case clause without indentation is more intuitive and is regarded as of better quality
And a line with more than 120 characters is too messy

Let's
*Remove the indentations in the code
*Split long lines into two

choiwab committed:
Merge pull request #3 from choiwab/A-CodeQuality

Adjust for Better Code Quality

choiwab committed:
Add reminder functionality for tasks due or occurring today

Currently, there is no functionality that alerts users about tasks (deadlines or events) that are due or occurring today. The Deadline and Event classes do not have a way to easily compare their dates with the current date.

Adding this reminder feature improves the user experience by notifying them of upcoming tasks, ensuring they stay on top of their schedule.

Implement getDate() methods in both Deadline and Event classes to extract the LocalDate from the LocalDateTime fields. Use these methods to compare task dates with the current date.

This approach is chosen to simplify the date comparison process without altering the existing structure of the classes. It ensures clean separation of concerns, making the date retrieval more modular and reusable.

These changes prepare the system for future enhancements, such as automated daily reminders and more comprehensive scheduling features.

choiwab committed:
Merge branch-Reminder to master

choiwab committed:
Adjust UI Blacknut Logo message

choiwab committed:
Add scrrenshot of product

choiwab committed:
Add user guide

choiwab committed:
Merge branch-A-UserGuide into master

choiwab committed:
Update README.md

choiwab committed:
Merge branch-A-UserGuide into master

choiwab committed:
Ui.java: add javadoc documents

choiwab committed:
Ui.java: add javadoc documents for getError() and getLine()

choiwab committed:
Todo.java: add javadoc documents

choiwab committed:
Event.java: remove blank line

choiwab committed:
Command.java: remove blank line

choiwab committed:
Ui.java: remove unused library

choiwab committed:
Todo.java: remove unused import statements

choiwab committed:
Storage.java: remove unused import statements

choiwab committed:
MainWindow.java: add line between import statement and class

choiwab committed:
Blacknut.java: remove unused import statements

choiwab committed:
ParserTest.java: Remove 3 tests

choiwab committed:
Merge branch branch-A-Release into master

choiwab committed:
Add Launcher.java
